public/app/plugins/datasource/cloudwatch/query-runner/CloudWatchLogsQueryRunner.ts

```diff
@@ -105,15 +105,15 @@ export class CloudWatchLogsQueryRunner extends CloudWatchRequest {
     });
 
     if (hasQueryWithMissingLogGroupSelection) {
-      return of({ data: [], error: { message: 'Log group is required' } });
+      return of({ data: [] });
     }
 
     const hasQueryWithMissingQueryString = queryParams.some((qp) => {
```

Comment:
I know this was not introduced by you in this or in the previous PR, but I don't like this assumption. If one out of many queries are incomplete, let stop all of them. It's better to filter out the broken ones and move further with the ones that are okay. 

Can you instead add a new method in this file called `filterQuery`? This method should check whether the logs query is valid for execution or not. Then in the `handleLogQueries` method, you can call it and exclude incomplete queries from further execution.
`const validLogsQueries = queryParams.filter(this.filterQuery)`
This is the pattern that the [data source api suggests](https://github.com/grafana/grafana/blob/main/packages/grafana-runtime/src/utils/DataSourceWithBackend.ts#L126-L128).  

The variable name `queryParams` does not really makes sense to me..? Can this be changed to something like this maybe? Then the filtering should be applied before this.
`const startQueryRequests: StartQueryRequest[] = logQueries.map((target: CloudWatchLogsQuery) => ...`

Reply:
Just a remark looking at the filter() method, might not be relevant: Are we ok with users not being informed they are sending an incomplete query and that it won't go through? 
I guess it's not that relevant here since there are only two mandatory fields, but the practice of filtering them could be confusing elsewhere

Reply:
That's a really valid concern! I think for now, we should put emphasis on improving the query editor form so that it becomes obvious to the user what fields are required and which ones are optional. For example, the log groups selector uses react select and we could set the `invalid` prop on blur in case no log groups were selected.